runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -245,31 +252,33 @@ void registerTable(@Nonnull String namespace,
              *    2.2.1: Existing protobuf file list matches existing proto files
              *    2.2.3: Existing protobuf file's descriptor mismatches with the incoming descriptor
+             *
+             *    Note that in all cases, the registryTable needs to be
+             *    updated with the table descriptors because the Any messages
+             *    for key, value and metadata were added to tableDescriptor in
+             *    0.3.2.  To populate these fields on upgraded environments,
+             *    registry table is always updated.
              */
             if (TransactionalContext.isInTransaction()) {
                 throw new IllegalThreadStateException("openTable: Called on an existing transaction");
             }
             try {
                 this.runtime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
                 CorfuRecord<TableDescriptors, TableMetadata> oldRecord = this.registryTable.get(tableNameKey);
-                if (oldRecord == null) { // Case 1 above, new unseen table
-                    this.registryTable.put(tableNameKey,
-                            new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
-                } else { // Case 2 above, re-open of existing table
-                    if (!oldRecord.getPayload().getFileDescriptorsMap().keySet()
-                            .equals(tableDescriptors.getFileDescriptorsMap().keySet())) {
-                        log.warn("Protobuf file list changed for {} table {}. Prev list {}, new list {}",
-                                namespace, tableName, oldRecord.getPayload().getFileDescriptorsMap().keySet(),
-                                tableDescriptors.getFileDescriptorsMap().keySet());
-                        for (StackTraceElement st : Thread.currentThread().getStackTrace()) {
-                            log.debug("{}", st);
-                        }
-                        this.registryTable.put(tableNameKey,
-                                new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
-                    } // else no need to re-add the exact same entry into the table
+                // Case 2 above, re-open of existing table
+                if (oldRecord != null && !oldRecord.getPayload().getFileDescriptorsMap().keySet()
+                    .equals(tableDescriptors.getFileDescriptorsMap().keySet())) {
+                    log.warn("Protobuf file list changed for {} table {}. Prev list {}, new list {}",
+                        namespace, tableName, oldRecord.getPayload().getFileDescriptorsMap().keySet(),
+                        tableDescriptors.getFileDescriptorsMap().keySet());
+                    for (StackTraceElement st : Thread.currentThread().getStackTrace()) {
+                        log.debug("{}", st);
+                    }
                 }
+                this.registryTable.put(tableNameKey,
```

Comment:
Let's try to avoid an unconditional IO since this is only needed if oldRecord != null && oldRecord.getPayload().getKey() == null
Basically only when the old record did not have any default instances recorded AKA upgrade

Reply:
but oldRecord.getPayload().getKey() will not be null.  It will be a default Any message, right?